backend/lib/cache/index.js

```diff
@@ -48,25 +77,34 @@ module.exports = {
     return cache.getSeeds()
   },
   getSeed (name) {
-    return _.cloneDeep(_.find(cache.getSeeds(), ['metadata.name', name]))
+    return _
+      .chain(cache.getSeeds())
+      .find(['metadata.name', name])
+      .cloneDeep()
+      .value()
   },
   getVisibleAndNotProtectedSeeds () {
     const predicate = item => {
-      const taints = _.get(item, 'spec.taints', [])
-      let seedProtected = false
-      let seedInVisible = false
-      _.forEach(taints, taint => {
-        if (taint.key === 'seed.gardener.cloud/protected') {
-          seedProtected = true
-        }
-        if (taint.key === 'seed.gardener.cloud/invisible') {
-          seedInVisible = true
-        }
-      })
-      return !seedProtected && !seedInVisible
+      const taints = _.get(item, 'spec.taints')
+      const unprotected = !_.find(taints, ['key', 'seed.gardener.cloud/protected'])
+      const visible = !_.find(taints, ['key', 'seed.gardener.cloud/invisible'])
+      return unprotected && visible
     }
     return _.filter(cache.getSeeds(), predicate)
   },
+  getProjects () {
+    return cache.getProjects()
+  },
+  findProjectByNamespace (namespace) {
```

Comment:
let's talk about this method. Maybe we need to add a disclaimer (because no subjectaccessreview is done)..

Reply:
That is the case for all methods of the cache. My proposal has been to introduce a module `entities` which wraps the access to kubernetes resources and reads from remote server or from cache and also does authorization check if it returns from cache.

Reply:
Added this comment
```js
  /*
    In file `lib/api.js` the synchronization is started with the privileged dashboardClient.
    Be careful when reading information from the cache that an authorization check is done
    or the information can be considered as not sensitive or public.
  */
```